src/components/file-input/file-input.spec.tsx

```diff
@@ -0,0 +1,380 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { rootTagTestRtl } from "../../__internal__/utils/helpers/tags/tags-specs/tags-specs";
+import { assertStyleMatch } from "../../__spec_helper__/test-utils";
+
+import FileInput, { FileUploadStatusProps } from ".";
+
+describe("rendering with no file uploaded", () => {
+  it("renders a hidden HTML file input element", () => {
+    render(<FileInput label="file input" onChange={() => {}} />);
+    const hiddenInput = screen.queryByLabelText("file input");
+    expect(hiddenInput).toBeInTheDocument();
+    expect(hiddenInput).not.toBeVisible();
+  });
+
+  it("renders an HTML button to choose a file to add", () => {
+    render(<FileInput onChange={() => {}} />);
+    expect(
+      screen.queryByRole("button", { name: "Select file" })
+    ).toBeInTheDocument();
+  });
+
+  it("accepts an accept prop and passes it to the underlying input", () => {
+    render(
+      <FileInput label="file input" accept="image/*,.pdf" onChange={() => {}} />
+    );
+    const hiddenInput = screen.queryByLabelText("file input");
+    expect(hiddenInput).toHaveAttribute("accept", "image/*,.pdf");
+  });
+
+  it("accepts a buttonText prop to change the button text", () => {
+    render(<FileInput buttonText="add a file" onChange={() => {}} />);
+    expect(
+      screen.queryByRole("button", { name: "add a file" })
+    ).toBeInTheDocument();
+  });
+
+  it("accepts a dragAndDropText prop to change the main text", () => {
+    render(<FileInput dragAndDropText="drop zone" onChange={() => {}} />);
+    expect(screen.queryByText("drop zone")).toBeInTheDocument();
+  });
+
+  it("accepts a label prop and renders it as a visible label", () => {
+    render(<FileInput label="file input" onChange={() => {}} />);
+    const label = screen.getByText("file input");
+    expect(label).toBeInTheDocument();
+    expect(label).toBeVisible();
+  });
+
+  it("accepts an inputHint prop", () => {
+    render(<FileInput inputHint="help" onChange={() => {}} />);
+    const hintText = screen.getByText("help");
+    expect(hintText).toBeInTheDocument();
+    expect(hintText).toBeVisible();
+  });
+
+  it("accepts data tag props", () => {
+    render(
+      <FileInput
+        data-element="element-test"
+        data-role="role-test"
+        onChange={() => {}}
+      />
+    );
+    const wrapperElement = screen.getByText("or drag and drop your file")
```

Comment:
Suggestion (non-blocking): might be worth storing this in a variable as it's used quite a lot, that way if the text is ever updated it only needs changing once in this file.

Reply:
Likewise happy to leave this as is and it can be addressed in the future if needed